### ISSUE TITLE
Images with captions not outputting caption text in transformed markup

### ISSUE THREAD

**Issue by matthewsimo** [Automattic/facebook-instant-articles-wp#475]
Captions on images aren't being transformed correctly and FB not approving IA Feed.
# Steps required to reproduce the problem

Given the following content entered for a post:

```
[caption id="attachment_25192" align="aligncenter" width="640"]<a href="http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549.jpg"><img class="wp-image-25192 size-large" src="http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549-860x1024.jpg" alt="FILE - In this Oct. 12, 2005, file photo, convicted BTK killer Dennis Rader listens during a court proceeding in El Dorado, Kan. (Travis Heying/The Wichita Eagle via AP, Pool)" width="640" height="762" /></a> FILE - In this October 12, 2005, file photo, convicted BTK killer Dennis Rader listens during a court proceeding in El Dorado, Kansas. (Travis Heying/<em>The Wichita Eagle</em> via AP, Pool)[/caption]
```

This corresponds to this source markup:

```
<div id="attachment_25192" style="width: 650px" class="wp-caption aligncenter"><a href="http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549.jpg"><img class="wp-image-25192 size-large" src="http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549-860x1024.jpg" alt="FILE - In this Oct. 12, 2005, file photo, convicted BTK killer Dennis Rader listens during a court proceeding in El Dorado, Kan. (Travis Heying/The Wichita Eagle via AP, Pool)" width="640" height="762" srcset="http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549-860x1024.jpg 860w, http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549-252x300.jpg 252w, http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549-768x914.jpg 768w" sizes="(max-width: 640px) 100vw, 640px" /></a><p class="wp-caption-text">FILE &#8211; In this October 12, 2005, file photo, convicted BTK killer Dennis Rader listens during a court proceeding in El Dorado, Kansas. (Travis Heying/<em>The Wichita Eagle</em> via AP, Pool)</p></div>
```

And the following is the transformed markup:

```
      <figure>
        <img src="http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549-860x1024.jpg"/>
      </figure>
```
# Expected Result

I would expect that the caption text would be added as well, just like in the FB docs - https://developers.facebook.com/docs/instant-articles/reference/caption#examples

Something like:

```
      <figure>
        <img src="http://crimefeed.com/wp-content/uploads/2016/09/AP_16232691572549-860x1024.jpg"/>
        <figcaption class="op-vertical-above">
                <h1>_caption text here_</h1>
        </figcaption>
      </figure>
```
# Actual Result

Caption text is completely ignored.

---

*\* Version Info **
- Plugin version: 3.1.3
- WordPress version: 4.6.1

**Comment by matthewsimo** [comment]
I was able to get this properly working but adding this custom transform rule:
```
    {
      "class": "CaptionRule",
      "selector": "p.wp-caption-text"
    }
```

I think this should be plug and play honestly though..

**Comment by diegoquinteiro** [comment]
Thanks for reporting!

This was fixed in 3.2 based on your feedback.
